services/cd-service/pkg/service/batch_test.go

```diff
@@ -26,10 +26,12 @@ import (
 
 	"github.com/freiheit-com/kuberpult/services/cd-service/pkg/testutil"
 	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/codes"
```

Comment:
overall this PR is fine, but it's really hard to test.
I discussed with @gsandok an approach to solve this.
Basically we will provide a way mock the authentication handler (dex), and overwrite with a role.
This way we can test this locally without having to setup dex.
(And my hidden agenda here is that this may be useful for technical users too).